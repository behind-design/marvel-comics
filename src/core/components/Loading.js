import React from 'react';
import ReactLoading from 'react-loading';
import cx from 'classnames';


export default ({ loading }) => (
  <div className={cx({
    'marvel-comics-loading': true,
    active: loading
  })}>
    <ReactLoading type="bars" color="#ED1D24" width="90px" />
  </div>
)
