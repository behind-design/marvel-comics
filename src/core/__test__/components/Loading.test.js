import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';

import Loading from '../../components/Loading';


const IMAGE_NOT_FOUND = `http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_uncanny.jpg`;

describe('Loading', () => {
  it('renders without crashing', () => {
    render(
      <Loading />
    );
  });

  it('renders all the correct properties', () => {
    const wrapper = shallow(
      <Loading />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).not.toHaveClassName('active');
  });

  it('sets the class to active when loading is true', () => {
    const wrapper = shallow(
      <Loading loading={true} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveClassName('active');
  });
});
