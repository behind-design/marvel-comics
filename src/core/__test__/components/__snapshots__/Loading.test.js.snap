// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders all the correct properties 1`] = `
<div
  className="marvel-comics-loading"
>
  <Loading
    color="#ED1D24"
    delay={1000}
    height={64}
    type="bars"
    width="90px"
  />
</div>
`;

exports[`Loading sets the class to active when loading is true 1`] = `
<div
  className="marvel-comics-loading active"
>
  <Loading
    color="#ED1D24"
    delay={1000}
    height={64}
    type="bars"
    width="90px"
  />
</div>
`;
